src/agent/agent.c

```diff
@@ -1808,6 +1808,43 @@ static int agent_method_remove_proxy(sd_bus_message *m, UNUSED void *userdata, U
 }
 
 
+/*************************************************************************
+ **** org.eclipse.bluechi.Agent.SwitchController ******
+ *************************************************************************/
+
+static int agent_method_switch_controller(sd_bus_message *m, void *userdata, UNUSED sd_bus_error *ret_error) {
+        Agent *agent = userdata;
+        const char *dbus_address = NULL;
+
+        int r = sd_bus_message_read(m, "s", &dbus_address);
+        if (r < 0) {
+                bc_log_errorf("Failed to read DbusAddress parameter: %s", strerror(-r));
+                return sd_bus_reply_method_errorf(
+                                m,
+                                SD_BUS_ERROR_FAILED,
+                                "Failed to read DbusAddress parameter: %s",
+                                strerror(-r));
+        }
+
+        if (!agent_set_controller_address(agent, dbus_address)) {
```

Comment:
This only sets agent->controller_address, but if ensure_orch_address has already run, then this has been copied to agent->orch_addr, (and that is what the property reads)  and this newly set value will never be used. So, I can't see how this works at all? 

What you want here is agent_set_orch_address, surely?

Reply:
Ah, i see that orch_addr is reset on reconnect, and we now always reconnect, so i guess this is fine.